velox/functions/lib/Re2Functions.cpp

```diff
@@ -970,51 +969,143 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures() {
   };
 }
 
-PatternMetadata determinePatternKind(StringView pattern) {
+std::string unescape(
+    StringView pattern,
+    vector_size_t start,
+    vector_size_t end,
+    std::optional<char> escapeChar) {
+  if (!escapeChar) {
+    return std::string(pattern.data(), start, end - start);
+  }
+
+  std::ostringstream str;
+  bool isPreviousEscapeChar = false;
+  for (auto index = start; index < end; index++) {
```

Comment:
nit: I wonder if it will be more efficient to use std::string::find(escapeChar) instead of looping over characters